File: frontend/src/types/employee.ts
import type { Role } from './common'

export interface Employee {
  id: string
  name: string
  email: string
  phone: string
  cpf: string
  role: Role
  active: boolean
  position: string
  hiredAt: string
  salary?: number
  createdAt: string
}

export interface EmployeePayload {
  name: string
  email: string
  password?: string
  role: Role
  phone: string
  position: string
  hiredAt: string
  salary?: number
  cpf: string
}

File: backend/src/main/java/com/gymflow/pro/entity/Employee.java
package com.gymflow.pro.entity;

import com.gymflow.pro.entity.enums.StudentStatus;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "employees")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;

    @Column(nullable = false, length = 100)
    private String position;

    @Column(name = "hired_at", nullable = false)
    private LocalDate hiredAt;

    private BigDecimal salary;

    @Column(length = 14, unique = true)
    private String cpf;

    @Builder.Default
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private StudentStatus status = StudentStatus.ACTIVE;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}
